cpp/algo_module/algorithm/algo.hpp

```diff
@@ -1,5 +1,220 @@
+    auto it = set.find(elem->node.Id());
+    if (it != set.end()) {
+      if (elem->total_distance < it->second) {
+        it->second = elem->total_distance;
+        pq.push(elem);
+      }
+      return;
+    }
+    pq.push(elem);
+    set.insert({elem->node.Id(), elem->total_distance});
+  }
+
+  Open() = default;
+};
+
+class Closed {
+ public:
+  std::unordered_set<std::shared_ptr<NodeObject>, NodeObject::Hash, NodeObject::Equal> closed;
+
+  bool Empty() { return closed.empty(); }
+  void Erase(const std::shared_ptr<NodeObject> &obj) { closed.erase(obj); }
+
+  bool FindAndCompare(const std::shared_ptr<NodeObject> &obj) {
+    auto it = closed.find(obj);
+    if (it != closed.end()) {
+      bool erase = (*it)->total_distance > obj->total_distance;
+      if (erase) {
+        closed.erase(it);
+      }
+      return erase;
+    }
+    return true;
+  }
+
+  void Insert(const std::shared_ptr<NodeObject> &obj) { closed.insert(obj); }
+
+  Closed() = default;
+};
+
+class Config {
+ public:
+  bool unweighted = false;
+  double epsilon = 1.0;
+  std::string distance_prop = "distance";
+  std::string heuristic_name = "";
+  std::string latitude_name = "lat";
+  std::string longitude_name = "lon";
+  std::unordered_set<std::string> whitelist;
+  std::unordered_set<std::string> blacklist;
+  std::unordered_set<std::string> in_rels;
+  std::unordered_set<std::string> out_rels;
+  bool duration = false;
+
+  Config(const mgp::Map &map) {
+    if (!map.At("unweighted").IsNull() && map.At("unweighted").IsBool()) {
+      unweighted = map.At("unweighted").ValueBool();
+    }
+    if (!map.At("epsilon").IsNull() && map.At("epsilon").IsNumeric()) {
+      epsilon = map.At("epsilon").ValueNumeric();
+    }
+    if (!map.At("distance_prop").IsNull() && map.At("distance_prop").IsString()) {
+      distance_prop = map.At("distance_prop").ValueString();
+    }
+    if (!map.At("heuristic_name").IsNull() && map.At("heuristic_name").IsString()) {
+      heuristic_name = map.At("heuristic_name").ValueString();
+    }
+    if (!map.At("latitude_name").IsNull() && map.At("latitude_name").IsString()) {
+      latitude_name = map.At("latitude_name").ValueString();
+    }
+    if (!map.At("longitude_name").IsNull() && map.At("longitude_name").IsString()) {
+      longitude_name = map.At("longitude_name").ValueString();
+    }
+    if (!map.At("whitelisted_labels").IsNull() && map.At("whitelisted_labels").IsList()) {
+      auto list = map.At("whitelisted_labels").ValueList();
+      for (auto value : list) {
+        if (value.IsString()) {
+          whitelist.insert(std::string(value.ValueString()));
+        }
+      }
+    }
+    if (!map.At("blacklisted_labels").IsNull() && map.At("blacklisted_labels").IsList()) {
+      auto list = map.At("blacklisted_labels").ValueList();
+      for (auto value : list) {
+        if (value.IsString()) {
+          blacklist.insert(std::string(value.ValueString()));
+        }
+      }
+    }
```

Comment:
I dont think that makes sense, because I dont throw errors for this one, my reasoning is I can allow users typos in labels list, for example, I wont throw error for label list ["Label", 2 , "Label3"], i will just ignore the 2

Reply:
Talked on meet, continue resolving